exercises/spiral/main.go

```diff
@@ -0,0 +1,87 @@
+package main
+
+import "fmt"
+
+func spiral(x, y int) [][]int {
+
+	s := make([][]int, y)
+	for i := range s {
+		s[i] = make([]int, x)
+	}
+
+	l := 1
+
+	a := 0
+	b := 0
+	xmin := 0
+	ymin := 0
+	xmax := x - 1
+	ymax := y - 1
+	for l < x*y {
+
+		for a <= xmax && b <= ymax {
+			s[b][a] = l
+			l++
+
+			a++
+		}
+		a--
+		ymin++
+		b++
+
+		for a <= xmax && b <= ymax {
+			s[b][a] = l
+			l++
+
+			b++
+		}
+		b--
+		xmax--
+		a--
+
+		for a >= xmin && b >= ymin {
+			s[b][a] = l
+			l++
+
+			a--
+		}
+		a++
+		ymax--
+		b--
+
+		for a >= xmin && b >= ymin {
+			s[b][a] = l
+			l++
+
+			b--
+		}
+		b++
+		xmin++
+		a++
+
+	}
+	if x == 1 && y == 1 {
+		s[0] = []int{1}
+	}
```

Comment:
Why do you need this special condition?

Reply:
If the height and width are 1, the function fills the rectangle with 0.